File: games/would-you-rather/frontend/types/room.ts
export interface Player {
  id: string;
  display_name: string;
  avatar: string;
  is_host: boolean;
  is_ready: boolean;
  is_spectator: boolean;
  connected: boolean;
}

export interface RecapEntry {
  option_a: string;
  option_b: string;
  votes: Record<string, "a" | "b">;
}

export interface GameState {
  phase: "lobby" | "question_open" | "revealed" | "game_over";
  round_number: number;
  total_questions: number;
  option_a: string | null;
  option_b: string | null;
  voted: string[];
  revealed_votes: Record<string, string> | null;
  recap: RecapEntry[] | null;
}

export interface Room {
  code: string;
  game_type: string;
  is_private: boolean;
  phase: string;
  max_players: number;
  players: Player[];
  invite_url: string;
  game_state: GameState | null;
}

File: games/would-you-rather/backend/app/schemas/room.py
from pydantic import BaseModel


class CreateRoomRequest(BaseModel):
    game_type: str
    host_display_name: str
    host_avatar: str = "default"
    is_private: bool = False


class JoinRoomRequest(BaseModel):
    display_name: str
    avatar: str = "default"
    as_spectator: bool = False


class PlayerOut(BaseModel):
    id: str
    display_name: str
    avatar: str
    is_host: bool
    is_ready: bool
    is_spectator: bool
    connected: bool


class GameStateOut(BaseModel):
    phase: str
    round_number: int
    total_questions: int = 0
    option_a: str | None = None
    option_b: str | None = None
    voted: list[str] = []
    revealed_votes: dict[str, str] | None = None
    recap: list[dict] | None = None


class RoomOut(BaseModel):
    code: str
    game_type: str
    is_private: bool
    phase: str
    max_players: int
    players: list[PlayerOut]
    invite_url: str
    game_state: GameStateOut | None = None


class CreateRoomResponse(BaseModel):
    room: RoomOut
    player_id: str


class RoomSummary(BaseModel):
    code: str
    game_type: str
    phase: str
    is_private: bool
    player_count: int
    max_players: int
